infrastructure/docker/Dockerfile-ray-qiskit-arm64

```diff
@@ -2,8 +2,9 @@ ARG IMAGE_PY_VERSION=py39
 FROM rayproject/ray:2.3.0-$IMAGE_PY_VERSION-aarch64
 USER $RAY_UID
 
-RUN apt-get -y update && apt-get -y install gcc build-essential libopenblas-dev
+RUN apt-get -y update && apt-get -y install gcc build-essential libopenblas-dev cmake
 COPY --chown=$RAY_UID:$RAY_UID ./client ./qs
 RUN cd ./qs && pip install .
+RUN pip uninstall pyscf && pip install git+https://github.com/pyscf/pyscf 
```

Comment:
Why are we uninstalling it? it should not be in this image in a first place :) 

Or your suggestion is to have it in this specific image for m1, so local dev/testing is possible with pyscf?

Reply:
I didn't check if it's there already or not. :-).  The one installed by `pip install pyscf` doesn't work so I uninstall it in case it's there.  

In this [comment](https://github.com/pyscf/pyscf/issues/1511#issuecomment-1313884842) 
`pip install git+https://github.com/pyscf/pyscf` must be used to install it.

Reply:
The **PySCFDriver** depends on **pyscf**.  It must be in the Ray node or installed on the fly.

Reply:
I looked into a little bit.  The built image doesn't have 'pyscf` module.  It is installed by Ray before it executes   `PySCFDriver` that depends on `pyscf`.  The `pyscf` module installed by Ray has an issue and fails the execution.  Ray doesn't install `pyscf` if it is pre-installed in the image.  This PR pre-installs the correct `pyscf` not to let Ray install the wrong `pyscf`.

Reply:
What I'm relatively worry about is that we are installing `pyscf` from the repo directly without a versioning. Did you try @akihikokuroda using the last one version, `2.2.1`?

Reply:
@Tansito I fixed the version of the `pyscf` to 2.1.1.